packages/utils/src/string.ts

```diff
@@ -112,42 +120,49 @@ export function isMatchingPattern(value: string, pattern: RegExp | string): bool
  * @returns A base64-encoded version of the string
  */
 export function unicodeToBase64(plaintext: string): string {
-  const global = getGlobalObject();
-
-  // Cast to a string just in case we're given something else
-  const stringifiedInput = String(plaintext);
-  const errMsg = `Unable to convert to base64: ${
-    stringifiedInput.length > 256 ? `${stringifiedInput.slice(0, 256)}...` : stringifiedInput
-  }`;
+  const globalObject = getGlobalObject<GlobalWithBase64Helpers>();
 
   // To account for the fact that different platforms use different character encodings natively, our `tracestate`
   // spec calls for all jsonified data to be encoded in UTF-8 bytes before being passed to the base64 encoder.
   try {
+    if (typeof plaintext !== 'string') {
+      throw new Error(`Input must be a string. Received input of type '${typeof plaintext}'.`);
+    }
+
     // browser
-    if ('btoa' in global) {
+    if ('btoa' in globalObject) {
       // encode using UTF-8
       const bytes = new TextEncoder().encode(plaintext);
 
       // decode using UTF-16 (JS's native encoding) since `btoa` requires string input
       const bytesAsString = String.fromCharCode(...bytes);
 
-      return btoa(bytesAsString);
+      // TODO: if TS ever learns about "in", we can get rid of the non-null assertion
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return globalObject.btoa!(bytesAsString);
     }
 
     // Node
-    if ('Buffer' in global) {
+    if ('Buffer' in globalObject) {
       // encode using UTF-8
-      const bytes = Buffer.from(plaintext, 'utf-8');
+      // TODO: if TS ever learns about "in", we can get rid of the non-null assertion
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const bytes = globalObject.Buffer!.from(plaintext, 'utf-8');
 
       // unlike the browser, Node can go straight from bytes to base64
       return bytes.toString('base64');
     }
+
+    // we shouldn't ever get here, because one of `btoa` and `Buffer` should exist, but just in case...
+    throw new SentryError('Neither `window.btoa` nor `global.Buffer` is defined.');
   } catch (err) {
+    // Cast to a string just in case we're given something else
+    const stringifiedInput = JSON.stringify(plaintext);
```

Comment:
Tests are failing due to this serialization. `JSON.stringify(undefined)` returns `undefined` which fails `.length` accessor.

(you also missed my `Object.prototype.toString` comment. resolved, but not applied :p)

Reply:
No, I didn't miss it: https://github.com/getsentry/sentry-javascript/pull/3372#discussion_r610883346

But you're right that I somehow missed running tests a final time before I pushed, which is unlike me. 😕  Fixed now.